src/GHC/Util/Unify.hs

```diff
@@ -224,6 +224,10 @@ unifyExp' nm root x@(L _ (HsApp _ x1 x2)) y@(L _ (HsApp _ y1 y2)) =
 unifyExp' nm root x y@(L _ (OpApp _ lhs2 op2@(L _ (HsVar _ op2')) rhs2)) =
   noExtra $ unifyExp nm root x y
 
+unifyExp' nm root (L _ (HsBracket _ (VarBr _ b0 (occNameStr -> v1))))
+                  (L _ (HsBracket _ (VarBr _ b1 (occNameStr -> v2))))
+    | b0 == b1 && isUnifyVar v1 = Just (Subst [(v1, strToVar v2)])
```

Comment:
Why does the RHS have to be a HsBracket/VarBr thing? Why can't it be any arbitrary expression? Or anything inside an HsBracket?

Reply:
Currently it means that things such as `''Maybe` match with `''a` on the LHS with `a` substituted into `Maybe`.

How should, say `1 + 2` match for `'a`? The rule `{lhs: "varE 'a", rhs: "[|a|]"}` will turn `varE (1 + 2)`, which is invalid code (unless `Name` had a `Num` instance), into the different and valid code `[|1 + 2|]`.

Reply:
Fair enough - I don't really know enough details about this, but I'll trust you.